examples/vector_search.py

```diff
@@ -0,0 +1,267 @@
+| title              + The title of the movie                                |
++--------------------+-------------------------------------------------------+
+| plot               | A short summary of the plot of the movie              |
++--------------------+-------------------------------------------------------+
+| fullplot           | A longer summary of the plot of the movie             |
++--------------------+-------------------------------------------------------+
+| cast               + A list of the names of the actors in the movie        |
++--------------------+-------------------------------------------------------+
+| genres             | A list of string values representing the different    |
+|                    | genres the movie belongs to                           |
++--------------------+-------------------------------------------------------+
+| runtime            | How long the move runs for in minutes                 |
++--------------------+-------------------------------------------------------+
+| poster             | A link to the poster for the movie                    |
++--------------------+-------------------------------------------------------+
+| languages          | A list of string values representing the different    |
+|                    | languages for the movie                               |
++--------------------+-------------------------------------------------------+
+| directors          | A list of the names of the directors of the movie     |
++--------------------+-------------------------------------------------------+
+| writers            | A list of the names of the writers of the movie       |
++--------------------+-------------------------------------------------------+
+
+This example uses the fullplot to create the vector embeddings for each
+movie. Other fields can be used by normal Coherence filters to further narrow
+down vector searches.
+
+Searching Vectors
+=================
+
+A common way to search data in Coherence caches is to use Coherence
+aggregators. The aggregator feature has been used to implement k-nearest
+neighbour (k-nn) vector searching using a new built-in aggregator named
+SimilaritySearch. When invoking a SimilaritySearch aggregator on a cache
+the results are returned as a list of QueryResult instances.
+
+The SimilaritySearch aggregator is used to perform a Knn vector search on a
+cache in the same way that normal Coherence aggregators are used.
+
+"""
+
+
+class MovieRepository:
+    """This class represents the repository of movies. It contains all the
+    code to load and search movie data."""
+
+    MODEL_NAME: Final[str] = "all-MiniLM-L6-v2"
+    """
+    This is a sentence-transformers model used for generating text embeddings.
+    See https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2
+    """
+
+    EMBEDDING_DIMENSIONS: Final[int] = 384
+    """Embedding dimension for all-MiniLM-L6-v2"""
```

Comment:
I think comments should be consistent with regards to ending punctuation.  Please add missing '.' and check others for consistency.